Improved the Instructions (#113)
zenml-io/zenml-projects: end-to-end-computer-vision/README.md

## ZenML

We recommend using our [ZenML Pro offering](https://cloud.zenml.io/) to get a
deployed instance of ZenML:

### Set up your environment

```bash
pip install -r requirements.txt

> zenml integration install torch gcp mlflow label-studio -y

Reviewer: ```suggestion
zenml integration install torch gcp mlflow label_studio -y
```

ZenML integrations have underscores, not hyphens.

```bash
pip install -r requirements.txt
zenml integration install torch gcp mlflow -y

> pip install label-studio

Reviewer: ```suggestion
```

I think this is no longer needed? And maybe you can add label_studio integration here too instead of below?

We will use GCP in the commands listed below, but it will work for other cloud
providers.

1) Follow our guide to set up your credentials for GCP [here](https://docs.zenml.io/how-to/auth-management/gcp-service-connector)

2) Set up a bucket in GCP to persist your training data

> 3) Set up a bucket to use as artifact store within ZenML
> Learn how to set up a GCP artifact store stack component within zenml

Reviewer: ```suggestion
3) Set up a bucket to use as artifact store within ZenML
Learn how to set up a GCP artifact store stack component within ZenML
```

[here](https://docs.zenml.io/stack-components/artifact-stores)

> 4) Set up vertex for pipeline orchestration

Reviewer: ```suggestion
4) Set up Vertex for pipeline orchestration
```

Learn how to set up a Vertex orchestrator stack component within zenml
[here](https://docs.zenml.io/stack-components/orchestrators/vertex)
5) For training on accelerators like GPUs/TPUs set up Vertex
Learn how to set up a Vertex step operator stack component within zenml
[here](https://docs.zenml.io/stack-components/step-operators/vertex)

> 6) Set up Container Registry in GCP. Learn how to set up a google cloud container registry component within zenml

Reviewer: ```suggestion
6) Set up a Container Registry in GCP. Learn how to set up a google cloud container registry component within ZenML
```

> Learn how to set up a Vertex orchestrator stack component within zenml
> [here](https://docs.zenml.io/stack-components/orchestrators/vertex)
> 5) For training on accelerators like GPUs/TPUs set up Vertex
> Learn how to set up a Vertex step operator stack component within zenml
> [here](https://docs.zenml.io/stack-components/step-operators/vertex)
> 6) Set up Container Registry in GCP. Learn how to set up a google cloud container registry component within zenml

Reviewer: ```suggestion
Learn how to set up a Vertex orchestrator stack component within ZenML
[here](https://docs.zenml.io/stack-components/orchestrators/vertex)
5) For training on accelerators like GPUs/TPUs set up Vertex
Learn how to set up a Vertex step operator stack component within zenml
[here](https://docs.zenml.io/stack-components/step-operators/vertex)
6) Set up Container Registry in GCP. Learn how to set up a google cloud container registry component within ZenML
```

[here](https://docs.zenml.io/stack-components/container-registries/gcp)

## Label Studio

1) [Start Label Studio locally](https://labelstud.io/guide/start)

> For label studio we recommend using docker/docker-compose to deploy a local instance

Reviewer: ```suggestion
For Label Studio we recommend using docker/docker-compose to deploy a local instance
```

```bash

> zenml integration install label_studio

Reviewer: ```suggestion
```

No need for this here if you did it above already.

Author: good point

git clone https://github.com/HumanSignal/label-studio.git
cd label-studio
docker-compose up -d # starts label studio at http://localhost:8080
```
2) [Follow these ZenML instructions to set up Label Studio as a stack component](https://docs.zenml.io/stack-components/annotators/label-studio#how-to-deploy-it)
3) Create a project within Label Studio and name it `ship_detection_gcp`
![img.png](_assets/project_creation_label_studio.png)
4) Configure your project to use `Object Detection with Bounding Boxes` as Labeling Setup
![img.png](_assets/labeling_setup.png)
In the following screen you now need to configure the labeling interface. This is where you define the different classes that you want to detect. In our case this should be a single `ship` class.
![img.png](_assets/labeling_interface.png)
Additionally you might want to allow users to zoom during labeling. This can be configured when you scroll down on this same screen.
6) [Set up Label Studio to use external storage](https://labelstud.io/guide/storage) 

> use the first bucket that you created for data persistence

Reviewer: ```suggestion
Use the first bucket that you created for data persistence
```

> ## Huggingface

Reviewer: ```suggestion
## Hugging Face
```

> This specific project relies on a dataset loaded from huggingface. As such a free huggingface account is needed.

Reviewer: ```suggestion
This specific project relies on a dataset loaded from Hugging Face. As such a free Hugging Face account is needed.
```
